src/app/forms/validators/custom-validators.ts

```diff
@@ -298,4 +298,17 @@ export class CustomValidators {
       return null;
     };
   };
+
+  static updateFunctionCode = (): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const vehicleFunctionCode = control.root.get('techRecord_functionCode');
+      if (control.dirty) {
+        if (control?.value === 'rigid') vehicleFunctionCode?.setValue('R', { emitEvent: false });
```

Comment:
You can avoid this using a map with types and thier function codes and someting like this 

vehicleFunctionCode?.setValue(functionCode.get(TYPE_VARIBALE), { emitEvent: false })